cve_bin_tool/parsers/python.py

```diff
@@ -28,16 +28,27 @@ def run_checker(self, filename):
                     "--report",
                     "-",
                     "--quiet",
-                ]
+                ],
+            )
+        except subprocess.CalledProcessError:
+            self.logger.error(
+                f"{filename} not scanned: pip --dry-run was unable to get package versions."
+            )
+            self.logger.error(
+                "pip version >= 22.2 is required to scan python requirements files"
```

Comment:
```suggestion
                "pip version >= 22.2 is required to scan Python requirements files."
```
We should not log this error every time, what if the scan wasn't successful for some other reason? After all, we can check the `pip` version and log only if `pip` is not new enough.

Reply:
Maybe we should still log the underlying error in addition to the user-friendly message? Unless we know for sure `pip` is too old, then there's no point.

Reply:
This error always arises due to an older version of pip I think because in newer versions, no such error arises.